### PR TITLE
Add no-flags, add short aliases, make aliases more consistent

### LINKED ISSUE #1738: Stage1 tool's `--write-parameters` flag is superflous
The stage1 tool has a flag `-write-paramters` that's not useful since it can only set `DataWriter.write_parameters` to True, which already is the default.

Flags are only useful if they set values other than the default.

There is also currently no nice way to deactivate writing parameters, so the flag should probably be `--no-parameters` instead.

We could also follow a common pattern and provide pairs of flags for everything, so I would propose to change

`--write-images`, `--write-paramters` to `--images`, `--no-images`, `--parameters`, `--no-parameters`.

### PR DESCRIPTION
**Pull request by nbiederbeck**
Fixes #1738

### CODE REVIEW

File: ctapipe/tools/display_dl1.py [hunk 1]
```diff
@@ -136,22 +136,19 @@ class DisplayDL1Calib(Tool):
...
-    aliases = Dict(
-        dict(
-            input="EventSource.input_url",
-            max_events="EventSource.max_events",
-            T="DisplayDL1Calib.telescope",
-            O="ImagePlotter.output_path",
+    aliases = {
+        ("i", "input"): "EventSource.input_url",
+        ("m", "max-events"): "EventSource.max_events",
+        "T": "DisplayDL1Calib.telescope",
```
Comment (maxnoe) [inline ctapipe/tools/display_dl1.py:142]:
```suggestion
        ("t", "telescope"): "DisplayDL1Calib.telescope",
```

File: ctapipe/tools/display_integrator.py [hunk 2]
```diff
@@ -226,24 +226,19 @@ class DisplayIntegrator(Tool):
...
-        dict(
-            f="EventSource.input_url",
-            max_events="EventSource.max_events",
-            E="DisplayIntegrator.event_index",
-            T="DisplayIntegrator.telescope",
-            C="DisplayIntegrator.channel",
+    aliases = {
+        ("i", "input"): "EventSource.input_url",
+        ("m", "max-events"): "EventSource.max_events",
+        "E": "DisplayIntegrator.event_index",
```
Comment (maxnoe) [inline ctapipe/tools/display_integrator.py:232]: e / event-index, t / telescope / C / channel (I think `c` is config, right`?

File: ctapipe/tools/display_dl1.py (hunk 1 above)
Comment (nbiederbeck) [reply ctapipe/tools/display_dl1.py:142]: Ahh yes, I wanted to ask if this would be a good change

File: ctapipe/tools/display_integrator.py [hunk 3]
```diff
@@ -226,24 +226,19 @@ class DisplayIntegrator(Tool):
...
+        "T": "DisplayIntegrator.telescope",
+        "C": "DisplayIntegrator.channel",
+    }
+    flags = {
+        "id": (
+            {"DisplayDL1Calib": {"use_event_index": True}},
```
Comment (maxnoe) [inline ctapipe/tools/display_integrator.py:238]: This should also be a flag, right?

File: ctapipe/tools/display_integrator.py (hunk 2 above)
Comment (nbiederbeck) [reply ctapipe/tools/display_integrator.py:232]: Will change it as proposed, yes `c` is config

File: ctapipe/tools/dl1_merge.py [hunk 4]
```diff
@@ -166,6 +149,34 @@ class MergeTool(Tool):
...
+            "Don't skip DL1/Event/Telescope and Simulation/Event/Telescope parameters in output",
+        ),
+        flag(
+            "skip-broken-files",
+            "MergeTool.skip_broken_files",
+            "Skip broken files instead of raising an error",
+            "Don't skip broken files instead of raising an error",
+        ),
+    ]
+    for f in _flags:
```
Comment (maxnoe) [inline ctapipe/tools/dl1_merge.py:178]: Why this for loop?

File: ctapipe/tools/display_integrator.py (hunk 3 above)
Comment (nbiederbeck) [reply ctapipe/tools/display_integrator.py:238]: Was not sure about that one, but yes, why not

**Review by maxnoe — CHANGES_REQUESTED** [review]
The for loop looks strange in the class body,

it should be possible to just define flags like this, right?

```
flags = {
    **flag(...),
    **flag(...),
   ...
}
```

Also, there are some boolean flags remaining, that are not setup using `flag`, can you transform those as well?

**Comment by nbiederbeck** [comment]
I thought the loop is the most elegant solution because I forgot about `**`

**Comment by nbiederbeck** [comment]
do we want a --no-progress flag?

**Comment by maxnoe** [comment]
Yes, I think that makes sense. In case we switch the default for example. Having all boolean flags consistent is a big plus I'd say.

**Comment by nbiederbeck** [comment]
That leads me to another question: the should the `(-f, --overwrite)` flags (and e.g. `(-q, --quiet)`) also become negated? But only the long forms, I suppose? (e.g. no `--no-f`)

**Comment by maxnoe** [comment]
shortform negation doesn't make sense, correct.  How does `flag` do shortforms?

**Comment by nbiederbeck** [comment]
afaik it doesn't. It can't handle short/long tuples, only strings. So I think we need to add a long yes/no flag and the short form we have to build ourselves

**Comment by maxnoe** [comment]
It doesn't. And it also does not make sure that only one of the flags is given.

**Comment by nbiederbeck** [comment]
What do you mean exactly? When I give `--flag --no-flag` it takes the last one. When I give `-l --long` it doesn't really matter

**Comment by maxnoe** [comment]
I somehow expected that traitlets would raise an error for `--flag --no-flag`

File: ctapipe/tools/tests/test_tools.py [hunk 5]
```diff
@@ -221,7 +221,7 @@ def test_display_integrator(tmp_path):
     assert (
         run_tool(
             DisplayIntegrator(),
-            argv=shlex.split(f"--f={GAMMA_TEST_LARGE} " "--max_events=1 "),
+            argv=shlex.split(f"--input={GAMMA_TEST_LARGE} " "--max_events=1 "),
```
Comment (maxnoe) [inline ctapipe/tools/tests/test_tools.py:224]: Could you just give the arguments as list? `[f"--input={GAMMA_TEST_LARGE}", "--max_events=1"]`? 

using shlex.split here is not what we want I think. We know, what are the single arguments that need to be quoted together.

Comment (nbiederbeck) [reply ctapipe/tools/tests/test_tools.py:224]: for all occurences in that file?

**Review by maxnoe — APPROVED** [review]

**Review by kosack — APPROVED** [review]

**Pull request merged**